compiler/rustc_codegen_cranelift/src/constant.rs

```diff
@@ -175,6 +175,17 @@ pub(crate) fn codegen_const_value<'tcx>(
                             fx.module.declare_data_in_func(data_id, &mut fx.bcx.func);
                         fx.bcx.ins().global_value(fx.pointer_type, local_data_id)
                     }
+                    GlobalAlloc::Type { .. } => {
+                        let data_id = data_id_for_alloc_id(
```

Comment:
data_id_for_alloc_id won't work as that pushes a `TodoItem::Alloc`, but you aren't handling `GlobalAlloc::Type` there. That is why I suggested copying and adapting `data_id_for_vtable`.